app/src/main/java/ar/net/rainbyte/resistorvaluecalculator/MainActivity.kt

```diff
@@ -22,19 +20,47 @@ class MainActivity : Activity() {
     }
 
     private lateinit var binding: ActivityMainBinding
+    private lateinit var resistorValueTemplate: String
 
+    var figure1: Int = 0
     var figure2: Int = 0
     var figure3: Int = 0
     var multiplier: Float = 0f
     var multiplierTag: String = ""
     var tolerance: Float = 0f
+    var temperature: Int = 0
+    var bands: Int = 4
 
     private fun updateResistorValue() {
-        val base = figure2 * 10 + figure3
+        val base = if (bands > 4) figure1 * 100 else 0 + figure2 * 10 + figure3
```

Comment:
conditional can be avoided if figure1 is set to zero for 4-band mode

Reply:
also the conditional as-is there is missing the curly braces, so base is calculated incorrectly